activerecord/lib/active_record/associations.rb

```diff
@@ -625,6 +630,14 @@ def association_instance_set(name, association)
       # Similarly, if any of the +before_remove+ callbacks throw an exception, the object
       # will not be removed from the collection.
       #
+      # Note: In order to trigger remove callbacks use any of following:
+      #   * +firm.clients.destroy_all+
+      #   * +firm.clients.destroy(c)+
```

Comment:
```suggestion
      # To trigger remove callbacks, you must use +destroy+ / +destroy_all+ methods. For example:
      #
      #   * <tt>firm.clients.destroy(client)</tt>
      #   * <tt>firm.clients.destroy(*clients)</tt>
      #   * <tt>firm.clients.destroy_all</tt>
```

Reply:
do we not want to flag this as a Note? I have left the `Note:` part in but incorporated your feedback 🙇

Reply:
`Note: ` does not get specially rendered, if that is what you mean.  I think there is a natural tendency to prefix new text with "Note: " rather than integrate it more thoroughly with existing text.  I know I've been guilty of that many times, so I was trying to avoid that here.  But I can see how the extra call-out might be helpful, since this was a point of confusion.